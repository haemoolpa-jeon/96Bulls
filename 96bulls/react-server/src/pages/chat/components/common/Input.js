import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import styles from './Input.module.scss';

class Input extends Component {
  handleChange = (e) => {
    const { onChange } = this.props;
    onChange(e);
  };

  render() {
    const {
      id, name, value, type, placeholder, className,
    } = this.props;
    return (
      <Fragment>
        <input
          id={id}
          name={name}
          value={value}
          type={type}
          placeholder={placeholder}
          onChange={this.handleChange}
          className={`${styles.input} ${className}`}
          autoComplete="off"
        />
        <label
          htmlFor={id}
          className={styles.label}
        >
          {placeholder}
        </label>
      </Fragment>
    );
  }
}

Input.propTypes = {
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  value: PropTypes.string,
  type: PropTypes.string.isRequired,
  placeholder: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  className: PropTypes.string,
};

export default Input;
