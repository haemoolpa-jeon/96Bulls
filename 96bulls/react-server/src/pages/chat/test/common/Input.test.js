import React from 'react';
import Input from '../../components/common/Input';
import {shallow} from 'enzyme/build';

describe('Input', () => {
  let component = null;
  const mockCallback = jest.fn(e => e);
  it('초기 렌더링이 문제없이 되야함', () => {
      component = shallow(<Input
      id="email"
      name="email"
      placeholder="Email"
      type="email"
      value="nick@naver.com"
      onChange={mockCallback}
    />);
  });

  it('초기 렌더링 스냅샷 일치함', () => {
    expect(component).toMatchSnapshot();
  });

  it('onChange 파라미터가 input 요소의 value 값과 같아야 함', () => {
    component.find('input').simulate('change', 'hello');
    expect(mockCallback.mock.calls[0][0]).toBe('hello');
  });
});
