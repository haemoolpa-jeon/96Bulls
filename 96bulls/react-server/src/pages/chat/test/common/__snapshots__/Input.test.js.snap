// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input 초기 렌더링 스냅샷 일치함 1`] = `
<Fragment>
  <input
    autoComplete="off"
    className="input undefined"
    id="email"
    name="email"
    onChange={[Function]}
    placeholder="Email"
    type="email"
    value="nick@naver.com"
  />
  <label
    className="label"
    htmlFor="email"
  >
    Email
  </label>
</Fragment>
`;
